src/sentry/workflow_engine/processors/data_condition_group.py

```diff
@@ -0,0 +1,82 @@
+from typing import Any
+
+from sentry.utils.function_cache import cache_func_for_models
+from sentry.workflow_engine.models import DataCondition, DataConditionGroup
+from sentry.workflow_engine.types import ProcessedDataConditionResult
+
+
+@cache_func_for_models(
+    [(DataConditionGroup, lambda group: (group.id,))],
+)
+def get_data_condition_group(data_condition_group_id: int) -> DataConditionGroup | None:
+    try:
+        group = DataConditionGroup.objects.get(id=data_condition_group_id)
+    except DataConditionGroup.DoesNotExist:
+        group = None
+    return group
+
+
+@cache_func_for_models(
+    [(DataCondition, lambda condition: (condition.condition_group_id,))],
+)
```

Comment:
I think that the `recalculate=False` is still relevant here. Typically, we will update all conditions for an alert at the same time. No need to trigger multiple refetches of this data.